pkg/cloudprovider/vsphereparavirtual/loadbalancer_test.go

```diff
@@ -44,385 +40,310 @@ import (
+	assert.Equal(t, exists, false)
+	assert.Error(t, err)
+}
 
-			It("should return error", func() {
-				Expect(exists).Should(Equal(false))
-				Expect(err).Should(HaveOccurred())
-			})
-		})
-	})
+func TestGetLoadBalancer_VMServiceCreated(t *testing.T) {
+	lb, _ := newTestLoadBalancer()
+	testK8sService := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testK8sServiceName,
+			Namespace: testK8sServiceNameSpace,
+		},
+	}
 
-	Describe("UpdateLoadBalancer", func() {
-		var (
-			err error
-		)
+	_, err := lb.EnsureLoadBalancer(context.Background(), testClustername, testK8sService, []*v1.Node{})
+	assert.Equal(t, vmservice.ErrVMServiceIPNotFound, err)
 
-		JustBeforeEach(func() {
-			err = lb.UpdateLoadBalancer(context.Background(), testClustername, testK8sService, []*v1.Node{})
-		})
+	_, exists, err := lb.GetLoadBalancer(context.Background(), testClustername, testK8sService)
+	assert.Equal(t, exists, true)
+	assert.NoError(t, err)
 
-		Context("when GetVMService failed", func() {
-			BeforeEach(func() {
-				testK8sService = &v1.Service{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      testK8sServiceName,
-						Namespace: testK8sServiceNameSpace,
-					},
-				}
-			})
-			It("should return error", func() {
-				// Error should be NotFound during the Get() call
-				Expect(err).Should(HaveOccurred())
-				Expect(apierrors.IsNotFound(err))
-			})
-		})
+	err = lb.EnsureLoadBalancerDeleted(context.Background(), testClustername, testK8sService)
+	assert.NoError(t, err)
+}
 
-		Context("when VMService update failed", func() {
-			BeforeEach(func() {
-				testK8sService = &v1.Service{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      testK8sServiceName,
-						Namespace: testK8sServiceNameSpace,
-					},
-				}
+func TestUpdateLoadBalancer_GetVMServiceFailed(t *testing.T) {
+	lb, _ := newTestLoadBalancer()
+	testK8sService := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testK8sServiceName,
+			Namespace: testK8sServiceNameSpace,
+		},
+	}
 
-				// Add the service with no ports
-				_, err := lb.EnsureLoadBalancer(context.Background(), testClustername, testK8sService, []*v1.Node{})
-				Expect(err).Should(Equal(vmservice.ErrVMServiceIPNotFound))
+	err := lb.UpdateLoadBalancer(context.Background(), testClustername, testK8sService, []*v1.Node{})
+	// Error should be NotFound during the Get() call
+	assert.Error(t, err)
+	assert.Equal(t, err.Error(), "VirtualMachineService not found")
+}
 
-				// Update the service definition to add ports
-				testK8sService.Spec = v1.ServiceSpec{
-					Ports: []v1.ServicePort{
-						{
-							Name:     "test-port",
-							Port:     80,
-							NodePort: 30900,
-							Protocol: "TCP",
-						},
+func TestUpdateLoadBalancer(t *testing.T) {
+	testCases := []struct {
+		name      string
+		expectErr bool
+	}{
+		{
+			name:      "when VMService update failed",
+			expectErr: true,
+		},
+		{
+			name:      "when VMService is updated",
+			expectErr: true,
```

Comment:
The original testcase doesn't expect error here.

Maybe add another variable called updateFunc, one return error one doesn't

Reply:
oops I should set expectErr to false, will update!